zebra-chain/src/block/tests/generate.rs

```diff
@@ -130,3 +156,64 @@ fn single_transaction_block(oversized: bool) -> Block {
         transactions,
     }
 }
+
+/// Implementation of block generation with one transaction and multiple transparent outputs
+fn single_transaction_block_many_outputs(oversized: bool) -> Block {
+    // Dummy input and output
+    let input =
+        transparent::Input::zcash_deserialize(&zebra_test::vectors::DUMMY_INPUT1[..]).unwrap();
+    let output =
+        transparent::Output::zcash_deserialize(&zebra_test::vectors::DUMMY_OUTPUT1[..]).unwrap();
+
+    // A block header
+    let header = block_header();
+
+    // Serialize header
+    let mut data_header = Vec::new();
+    header
+        .zcash_serialize(&mut data_header)
+        .expect("Block header should serialize");
+
+    // Serialize a LockTime
+    let lock_time = LockTime::Time(DateTime::<Utc>::from_utc(
+        NaiveDateTime::from_timestamp(61, 0),
+        Utc,
+    ));
+    let mut data_locktime = Vec::new();
+    lock_time
+        .zcash_serialize(&mut data_locktime)
+        .expect("LockTime should serialize");
+
+    // Calculate the number of outputs we need
+    let mut max_outputs_in_tx = (MAX_BLOCK_BYTES as usize
+        - data_header.len()
+        - zebra_test::vectors::DUMMY_INPUT1[..].len()
+        - data_locktime.len())
+        / (zebra_test::vectors::DUMMY_OUTPUT1[..].len() - 1);
+
+    if oversized {
+        max_outputs_in_tx += 1;
+    }
+
+    // 1 single input
+    let inputs = vec![input];
+
+    // Create outputs to be just below the limit
+    let outputs = std::iter::repeat(output)
+        .take(max_outputs_in_tx)
+        .collect::<Vec<_>>();
+
+    // Create a big transaction
+    let big_transaction = Transaction::V1 {
+        inputs,
+        outputs,
+        lock_time,
+    };
+
+    // Put the big transaction into a block
+    let transactions = vec![Arc::new(big_transaction)];
+    Block {
+        header,
+        transactions,
+    }
```

Comment:
Idea: should we add an assertion here? Something like:

```suggestion
    let block = Block {
        header,
        transactions,
    };

    assert_eq!(oversized, block.zcash_serialize_to_vec().len() > MAX_BLOCK_BYTES);
    
    block
```

Reply:
Fixed in [fe033cf](https://github.com/ZcashFoundation/zebra/pull/3759/commits/fe033cf255508dcaeec206fd9aacfcb4ac8d1988).

The under-sized blocks were far too small, so this required a significant workaround for the test vectors, and multiple fixes to the size calculations.

I'm happy with the workarounds now, hope you are too!